x/liquidvesting/types/schedule.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import (
+	"errors"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkvesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	vestingTypes "github.com/haqq-network/haqq/x/vesting/types"
+)
+
+// SubtractAmountFromPeriods subtracts coin amount from given periods proportionally,
+// returns decreased periods and diff of initial periods and decreased periods
+func SubtractAmountFromPeriods(
+	minuendPeriods sdkvesting.Periods,
+	subtrahend sdk.Coin,
+) (decreasedPeriods, diffPeriods sdkvesting.Periods, err error) {
```

Comment:
```suggestion
) (decreasedPeriods, diffPeriods sdkvesting.Periods, err error) {
if len(minuendPeriods) == 0 {
 return [], [], nil
 }

```

Reply:
add testcast for this

Reply:
After further investigation i understand it is not quite correct. If minuendPeriods are empty array their sum is equal to zero and if subtrahend is greater than zero and subtracted from zero sum it should lead to error. Empty arrays should be returned only in case if subtrahend is equal to zero